### PR TITLE
EWPP-2246: ePoetry contact field type.

### PR DESCRIPTION
**Pull request by 22Alexandra**

### CODE REVIEW

File: modules/oe_translation_epoetry/src/Plugin/Field/FieldType/ContactItem.php [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+ *   default_widget = "oe_transalation_epoetry_contact_widget",
+ *   default_formatter = "oe_transalation_epoetry_contact_formatter"
+ * )
+ */
+class ContactItem extends FieldItemBase implements ContactItemInterface {
+
+  /**
+   * Value for the 'contact_type' setting: Requester.
+   */
+  const REQUESTER = 'Requester';
```
Comment (upchuk) [inline modules/oe_translation_epoetry/src/Plugin/Field/FieldType/ContactItem.php:27]: If you made an interface, you can go ahead and put these constants on the interface.

File: modules/oe_translation_epoetry/src/Plugin/Field/FieldType/ContactItem.php [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+      ($contact === NULL || $contact === '');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function propertyDefinitions(FieldStorageDefinitionInterface $field_definition) {
+    $properties['contact'] = DataDefinition::create('string')
+      ->setLabel(t('Contact'))
+      ->setDescription(t('The contact information about the Poetry request.'))
```
Comment (upchuk) [inline modules/oe_translation_epoetry/src/Plugin/Field/FieldType/ContactItem.php:67]: ePoetry

File: modules/oe_translation_epoetry/src/Plugin/Field/FieldWidget/ContactWidget.php [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+ *   field_types = {"oe_transalation_epoetry_contact"},
+ * )
+ */
+class ContactWidget extends WidgetBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function formElement(FieldItemListInterface $items, $delta, array $element, array &$form, FormStateInterface $form_state) {
+    $types = $items[$delta]->contactTypes();
```
Comment (upchuk) [inline modules/oe_translation_epoetry/src/Plugin/Field/FieldWidget/ContactWidget.php:26]: $items[$delta] is always there?

**Review by upchuk — CHANGES_REQUESTED** [review]

File: modules/oe_translation_epoetry/src/Plugin/Field/FieldWidget/ContactWidget.php (hunk 3 above)
Comment (22Alexandra) [reply modules/oe_translation_epoetry/src/Plugin/Field/FieldWidget/ContactWidget.php:26]: Yes.

**Review by upchuk — APPROVED** [review]

**Pull request merged**